dataflux_core/fast_list.py

```diff
@@ -184,7 +186,7 @@ def run(self) -> None:
                         and blob.storage_class in self.allowed_storage_classes
                     ):
                         self.results.add((blob.name, blob.size))
-                    self.start_range = blob.name
+                    self.start_range = blob.name.removeprefix(self.prefix)
```

Comment:
This seems surprising, especially since the prefix is added in the lines of code above. Can you add a comment explaining it?

Reply:
Still not sure I understand this. Why do we want the range calculation to be prefix-agnostic? Won't things work the same if we include the prefix in the start range that is passed to the range splitter?